compile/constant_value.go

```diff
@@ -143,6 +255,29 @@ func (c ConstantMap) Link(scope Scope) (ConstantValue, error) {
 	return ConstantMap(items), nil
 }
 
+// ConstantSet represents a set of constant values from the Thrift file.
+type ConstantSet []ConstantValue
+
+// Link for ConstantSet.
+func (c ConstantSet) Link(scope Scope, t TypeSpec) (ConstantValue, error) {
+	s, ok := t.(*SetSpec)
```

Comment:
The problem with that is that now ConstantList needs to have more special handling for when TypeSpec is a `*SetSpec` rather than just delegating to ConstantSet.